integration_tests/src/main/scala/com/nvidia/spark/rapids/tests/common/BenchUtils.scala

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.nvidia.spark.rapids.tests.common
+
+import java.io.{File, FileOutputStream}
+import java.time.Instant
+import java.util.concurrent.TimeUnit.NANOSECONDS
+
+import scala.collection.mutable.ListBuffer
+
+import org.json4s.DefaultFormats
+import org.json4s.jackson.JsonMethods.parse
+import org.json4s.jackson.Serialization.writePretty
+
+import org.apache.spark.{SPARK_BUILD_USER, SPARK_VERSION}
+import org.apache.spark.sql.{DataFrame, Row, SparkSession}
+
+object BenchUtils {
+
+  /**
+   * Run the specified number of cold and hot runs and record the timings and summary of the
+   * query and results to file, including all Spark configuration options and environment
+   * variables.
+   *
+   * @param spark The Spark session
+   * @param queryRunner Function to create a DataFrame from the Spark session.
+   * @param queryDescription A query name to be written to the JSON file.
+   * @param filenameStub The prefix the for the JSON file. The current timestamp will be appended
+   *                     to ensure that filenames are unique and that results are not inadvertently
+   *                     overwritten.
+   * @param numColdRuns The number of cold runs.
+   * @param numHotRuns The number of hot runs.
+   * @param maxResultsToRecord The maximum number of result rows to collect and store in the
+   *                           output JSON file. This can be used as a sanity check to compare
```

Comment:
Could there be another option to store the results in a file that is not the jason file?

Maybe that's a stretch.. but it would be something like "myTest-[timestamp].result", that way we can take that output as is and compare different runs for accuracy.

Reply:
Sure, it would be easy to write two files ... a summary file and a results file ... and I'm also thinking about changes to supporting writing out to parquet.

Reply:
The benchmark no longer attempts to capture output from the queries but has specific methods for collecting vs writing to csv/parquet. We can run a separate utility afterwards for verifying that results are correct.